File: bin/Data/AUI/Periodic/Sources/code_abutton.cpp
// AButton application source code
#include "PeriodicApp.h"

void PeriodicApp::setup_abutton( AWidget *layout)
{
    PODVector<AWidget*> dest;
    layout->SearchWidgetClass( "TBButton", dest );
    for (unsigned ii = 0; ii < dest.Size(); ii++)
        SubscribeToEvent(dest[ii], E_WIDGETEVENT, URHO3D_HANDLER(PeriodicApp, HandleAbuttonEvent ));
}

void PeriodicApp::HandleAbuttonEvent(StringHash eventType, VariantMap& eventData)
{
    using namespace WidgetEvent;
    AWidget* widget = static_cast<AWidget*>(eventData[P_TARGET].GetPtr());
    if ( widget == NULL ) return;
    if (eventData[P_TYPE] == UI_EVENT_TYPE_CLICK)
    {
        if (widget->GetId() == "uibuttoncode" )
        {
            AppLog( "AButton support : " + widget->GetId() + " was pressed " );
            ViewCode ( "Sources/code_abutton.cpp", widget->GetParent() );
        }
        if (widget->GetId() == "uibuttonlayout" )
        {
            AppLog( "AButton support : " + widget->GetId() + " was pressed ");
            ViewCode ( "Scenes/layout_abutton.ui.txt", widget->GetParent() );
        }

        if (widget->GetId() == "demobutton" )
        {
            AppLog( "AButton action : " + widget->GetId() + " was pressed ");
        }
        if (widget->GetId() == "buttonducky" )
        {
            AppLog( "AButton action : " + widget->GetId() + " was pressed ");
        }
        if (widget->GetId() == "buttonready" )
        {
            AppLog( "AButton action : " + widget->GetId() + " was pressed ");
        }
        if (widget->GetId() == "buttonatomic" )
        {
            AppLog( "AButton action : " + widget->GetId() + " was pressed ");
        }
        if (widget->GetId() == "buttongreen" )
        {
            AppLog( "AButton action : " + widget->GetId() + " was pressed ");
        }
    }
}

File: bin/Data/AUI/Periodic/LuaScripts/code_abutton.lua
-- AButton application source code

function setup_uibutton( layout)
    local dest = layout:SearchAWidgetClass( "TBButton" )
    for i, awid in ipairs(dest) do
        SubscribeToEvent(awid, "WidgetEvent", "HandleUibuttonEvent")
	end
end

function HandleUibuttonEvent( eventType, eventData )

    local widget = eventData["Target"]:GetPtr("AWidget")
    if widget == nil then
 		return;
	end
    if eventData["Type"]:GetInt() == UI_EVENT_TYPE_CLICK then
        if widget:GetId() == "demobutton" then
            AppLog( "AButton action : " .. widget:GetId() .. " was pressed ")
        end
        if widget:GetId() == "uibuttoncode" then
            AppLog( "AButton support : " .. widget:GetId() .. " was pressed " )
            ViewCode ( "LuaScripts/code_abutton.lua", widget:GetParent() );
        end
        if widget:GetId() == "uibuttonlayout" then
            AppLog( "AButton support : " .. widget:GetId() .. " was pressed ")
            ViewCode ( "Scenes/layout_abutton.ui.txt", widget:GetParent() )
        end
        if widget:GetId() == "buttonducky" then
            AppLog( "AButton action : " .. widget:GetId() .. " was pressed ")
        end
        if widget:GetId() == "buttonready" then
            AppLog( "AButton action : " .. widget:GetId() .. " was pressed ")
        end
        if widget:GetId() == "buttonatomic" then
            AppLog( "AButton action : " .. widget:GetId() .. " was pressed ")
        end
        if widget:GetId() == "buttongreen" then
            AppLog( "AButton action : " .. widget:GetId() .. " was pressed ")
        end
    end
end

